File: old/pythonTest/leetcode.py
import requests
import json


def getLeetcodeInformation(username: str) -> dict:
    # consts
    QUERYS = {
        # queryName: {path, query}
        "calendar": {
            "path": "/graphql/noj-go",
            "query": "query userProfileCalendar($userSlug: String!, $year: Int){userCalendar(userSlug: $userSlug, year: $year){submissionCalendar}}",
        },
        "recentACSubmissions": {
            "path": "/graphql/noj-go",
            "query": "query recentAcSubmissions($userSlug: String!){recentACSubmissions(userSlug: $userSlug) {submissionId\nsubmitTime\nquestion{\ntranslatedTitle\ntitleSlug\nquestionFrontendId}}}",
        },
        "rank": {
            "path": "/graphql/noj-go",
            "query": "query userContestRankingInfo($userSlug: String!){userContestRanking(userSlug: $userSlug) {attendedContestsCount\nrating\nglobalRanking\nlocalRanking\nglobalTotalParticipants\nlocalTotalParticipants\ntopPercentage}}",
        },
        "solvedProblem": {
            "path": "/graphql",
            "query": "query userQuestionProgress($userSlug: String!) {userProfileUserQuestionProgress(userSlug: $userSlug) {numAcceptedQuestions{\ndifficulty\ncount}numFailedQuestions{difficulty\ncount}numUntouchedQuestions{difficulty\ncount}}}"
        }
    }
    BASE_URL = "https://leetcode.cn"
    HEADERS = {
        'content-type': 'application/json'
    }

    result = dict()
    for queryName, query in QUERYS.items():
        payload = json.dumps({
            "query": query["query"],
            "variables": {
                "userSlug": username
            }
        })
        response = requests.request(
            "POST", BASE_URL + query["path"], headers=HEADERS, data=payload)
        print(response.apparent_encoding)
        result[queryName] = response.json()["data"]

    return result


print(getLeetcodeInformation("shenghaowu"))

File: backend/dataFetching/leetcode.js
import fetch, { Headers } from "node-fetch"
const userName = "shenghaowu"

var myHeaders = new Headers()
myHeaders.append("content-type", "application/json")

var graphql = JSON.stringify({
    query: `query getRecentSubmissionList($username: String!) {
	recentSubmissions(userSlug: $username) {
		id submitTime status question {
			title difficulty titleSlug topicTags{
				name
			}
		}
	}
}`,
    variables: { username: userName }
})
var requestOptions = {
    method: "POST",
    headers: myHeaders,
    body: graphql
}

fetch("https://leetcode.cn/graphql/", requestOptions)
    .then((response) => response.text())
    .then((result) => console.log(result))
    .catch((error) => console.log("error", error))